### PR TITLE
Add Unrequested Configurations table

### PR DESCRIPTION
**Pull request by toddburnside**
Creates the new `Unrequested Coordinates + Configurations + Constraints` table which allows the user to create new configuration requests. It also abstracts out the common columns with the `Requested Coordinates + Configurations + Constraints` table.

The next PR will save the states of these 2 tables and the observation validations table, as well as remove the ability to create a new configuration request from the observations validations table (it turns out to be not such a great place to make the requests from).

image
image

### CODE REVIEW

File: explore/src/main/scala/explore/model/AppContext.scala [hunk 1]
```diff
@@ -53,6 +55,27 @@ case class AppContext[F[_]](
   def replacePage(appTab: AppTab, programId: Program.Id, focused: Focused): Callback =
     setPageVia(appTab, programId, focused, SetRouteVia.HistoryReplace)
 
+  def routingLink(
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/model/AppContext.scala:58]: Good idea! There are a number of places where this could be used.

Comment (rpiaggio) [reply explore/src/main/scala/explore/model/AppContext.scala:58]: I see you have already refactored to use it.

Comment (toddburnside) [reply explore/src/main/scala/explore/model/AppContext.scala:58]: I refactored some of the areas, but I probably missed some.

**Comment by toddburnside** [comment]
> Screenshot?

Added to the description.

I should also add that the text entered in the request editor is currently ignored, but the message should be added to the API at some point soon.

**Pull request merged**